cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -315,3 +319,34 @@ func GetConfigMapWithRetry(client clientset.Interface, namespace, name string) (
 	}
 	return nil, lastError
 }
+
+// getConfigMapFromList uses List() to obtain a ConfigMap in a namespace with a timeout.
+// Currently client-go and metav1.GetOptions do not support timeouts, which means that Get() calls
+// can block for a few minutes in case the API server is not reachable ("i/o timeout" errors).
+//
+// To workaround that use List() with a FieldSelector, a Limit of 1 and a TimeoutSeconds of 1.
+//
+// TODO: A long term plan for client-go API calls is to support golang Context for cancelation and timeout.
+// Make sure we remove this workaround once this feature is supported.
+func getConfigMapFromList(client clientset.Interface, namespace, name string) (*v1.ConfigMap, error) {
+	listOptions := metav1.ListOptions{
+		FieldSelector:  fields.OneTermEqualSelector("metadata.name", name).String(),
+		Limit:          1,
+		TimeoutSeconds: utilpointer.Int64Ptr(1),
+	}
+	cms, err := client.CoreV1().ConfigMaps(namespace).List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+	// Even if Limit is set to 1, the list of items can end up being more that 1.
+	// Iterate trough the items and match the ConfigMap name.
```

Comment:
isn't `Limit` supported for ConfigMaps ?

Reply:
with a fake client, i'm seeing > 1 results, even if limit == 1.